pilot/pkg/xds/ads.go

```diff
@@ -302,8 +302,17 @@ func (s *DiscoveryServer) shouldRespond(con *Connection, request *discovery.Disc
 		return false
 	}
 
+	if len(request.ResourceNames) == 0 && !isWildcardTypeURL(request.TypeUrl) {
```

Comment:
May be add a small comment on when this can happen?

Reply:
I see that you have added very good commentary on isWildCardTypeURL. May be we should change that to shouldUnsubscribe and pass in the request there and let it decide? It may be more readable